torch_geometric/datasets/stochastic_blockmodel_dataset.py

```diff
@@ -0,0 +1,179 @@
+from torch_geometric.data import InMemoryDataset, Data
+from torch_geometric.utils import stochastic_blockmodel_graph
+
+
+class StochasticBlockModelDataset(InMemoryDataset):
+
+    def __init__(self, root,
+                 block_sizes: Union[List[int], Tensor],
+                 edge_probs: Union[List[List[float]], Tensor],
+                 num_channels: Optional[int] = None,
+                 is_undirected: bool = True,
+                 x_kws: Optional[dict] = None,
+                 transform=None, pre_transform=None):
+        r"""The synthetic graph dataset generated by the stochastic block
+        model. The node features of each block are sampled from normal
+        distributions where the centers of clusters are vertices of a
+        hypercube by :obj:`'sklearn.datasets.make_classification'` method.
+
+        Args:
+            root (string): Root directory where the dataset should be saved.
+            block_sizes ([int] or LongTensor): The sizes of blocks.
+            edge_probs ([[float]] or FloatTensor): The density of edges going
+                from each block to each other block. Must be symmetric if the
+                graph is undirected.
+            num_channels (int, optional): The number of node features. If given
+                as :obj:`None`, node features are not generated.
+            is_undirected (bool): Whether the graph to generate is undirected.
+            x_kws (dict, optional): The keyword arguments that are passed down
+                to :obj:`'sklearn.datasets.make_classification'` method in
+                drawing node features.
+            transform (callable, optional): A function/transform that takes in
+                an :obj:`torch_geometric.data.Data` object and returns a
+                transformed version. The data object will be transformed before
+                every access. (default: :obj:`None`)
+            pre_transform (callable, optional): A function/transform that takes
+                in an :obj:`torch_geometric.data.Data` object and returns a
+                transformed version. The data object will be transformed
+                before being saved to disk. (default: :obj:`None`)
+        """
+        if not torch.is_tensor(block_sizes):
+            block_sizes = torch.tensor(block_sizes, dtype=torch.long)
+        if not torch.is_tensor(edge_probs):
+            edge_probs = torch.tensor(edge_probs, dtype=torch.float)
+
+        self.block_sizes = block_sizes
+        self.edge_probs = edge_probs
+        self.num_channels = num_channels
+        self.is_undirected = is_undirected
+
+        self.x_kws = {'n_informative': num_channels,
+                      'n_redundant': 0,
+                      'flip_y': 0.0,
+                      'shuffle': False}
+        self.x_kws.update(x_kws or {})
+
+        super().__init__(root, transform, pre_transform)
+        self.data, self.slices = torch.load(self.processed_paths[0])
+
+    @property
+    def name(self):
```

Comment:
How fast is the generation of SBMs? Do we really need to save them or can we simply generate them on-the-fly?

Reply:
It does not take a long time if the size is similar to the benchmark we often use. However, since the generation process relies on the stochastic process, I think saving processed files is more appropriate than using them on the fly.